node-addon-api/Napi/napi.h

```diff
@@ -131,6 +131,27 @@ struct NapiValueRef : JsValueRef {
     virtual bool GetBoolean(bool defVal) const override { return isNull() ? defVal : m_napiVal.ToBoolean(); }
     virtual int32_t GetInt(int32_t defVal) const override { return isNumeric() ? m_napiVal.ToNumber() : defVal; }
     virtual uint32_t GetUInt(uint32_t defVal) const override { return isNumeric() ? m_napiVal.ToNumber() : defVal; }
+    virtual int64_t GetInt64(int64_t defVal) const override {
+        if (isNumeric()) {
+            double number = m_napiVal.ToNumber();
+            if (number >= std::numeric_limits<int64_t>::min() && number <= std::numeric_limits<int64_t>::max()) 
+                return static_cast<int64_t>(number);
+        }
+        return defVal;
+    }
+    virtual uint64_t GetUInt64(uint64_t defVal) const override { 
+        if (isNumeric()) {
+            double number = m_napiVal.ToNumber();
+            if (number >= std::numeric_limits<uint64_t>::min() && number <= std::numeric_limits<uint64_t>::max()) {

```

Comment:
Reject if < 0 or >= Number.MAX_SAFE_INTEGER.

Reply:
Made that change